src/Symfony/Component/Translation/Translator.php

```diff
@@ -372,6 +372,39 @@ private function doLoadCatalogue($locale)
                 $this->catalogues[$locale]->addCatalogue($this->loaders[$resource[0]]->load($resource[1], $locale, $resource[2]));
             }
         }
+
+        $this->resolveCatalogueMessages($this->catalogues[$locale]);
+    }
+
+    private function resolveCatalogueMessages($catalogue)
+    {
+        $expr = '{(.*?)}';
```

Comment:
Can you really use any symbols for a translation key? Shouldn't we restrict the list of symbols?